### PR TITLE
Breadcrumb improvements

### PR DESCRIPTION
**Pull request by christianmemije**
## Summary

* Addressing comments left in #1632

### CODE REVIEW

File: kolibri/core/assets/src/views/breadcrumbs/index.vue [hunk 1]
```diff
@@ -117,56 +137,53 @@
...
+              updatedCrumb.collapsed = true;
+              this.crumbs.splice(index, 1, updatedCrumb);
+            });
+            break;
           }
-        });
+
+          remainingWidth -= lastCrumbWidth;
+          const lastCrumb = tempCrumbs.pop();
+          lastCrumb.collapsed = false;
```
Comment (christianmemije) [inline kolibri/core/assets/src/views/breadcrumbs/index.vue:184]: Only sets collapsed to false if needed.

File: kolibri/core/assets/src/views/breadcrumbs/index.vue [hunk 2]
```diff
@@ -1,43 +1,64 @@
...
+
+    <div class="breadcrumbs breadcrumbs-offscreen">
+      <ol class="breadcrumbs-visible-items">
+        <template v-for="(crumb, index) in crumbs">
+          <li
+            v-if="index !== crumbs.length - 1"
+            :ref="`crumb${index}`"
+            class="breadcrumbs-visible-item breadcrumbs-visible-item-notlast"
+          >
+            <router-link :to="crumb.link" tabindex="-1">{{ crumb.text }}</router-link>
```
Comment (christianmemije) [inline kolibri/core/assets/src/views/breadcrumbs/index.vue:49]: Added tabindex to avoid tabbing, Any other markup I should add to make it `inaccessible`? :sweat_smile:

File: kolibri/core/assets/src/views/breadcrumbs/index.vue [hunk 3]
```diff
@@ -1,43 +1,64 @@
...
-              <router-link :to="crumb.link" :style="{ maxWidth: `${collapsedCrumbMaxWidth}px` }">
-                {{ crumb.text }}
-              </router-link>
-            </li>
-          </ol>
-        </div>
-      </ui-icon-button>
+  <div>
+    <nav class="breadcrumbs">
+      <div v-if="collapsedCrumbs.length" class="breadcrumbs-dropdown-wrapper">
```
Comment (indirectlylit) [inline kolibri/core/assets/src/views/breadcrumbs/index.vue]: this might be better as a `v-show` since the condition is liable to change at runtime

File: kolibri/core/assets/src/views/breadcrumbs/index.vue [hunk 4]
```diff
@@ -1,43 +1,64 @@
...
+            class="breadcrumbs-visible-item breadcrumb-visible-item-last"
+          >
+            <span :style="{ maxWidth: `${lastCrumbMaxWidth}px` }">{{ crumb.text }}</span>
+          </li>
+        </template>
+      </ol>
+    </nav>
+
+
+    <div class="breadcrumbs breadcrumbs-offscreen">
```
Comment (indirectlylit) [inline kolibri/core/assets/src/views/breadcrumbs/index.vue]: Might want to add a comment to this div for future devs to understand why it exists.

Also, we should add `aria-hidden="true"` to this div - see [effects](https://www.paciellogroup.com/blog/2012/05/html5-accessibility-chops-hidden-and-aria-hidden/).

File: kolibri/core/assets/src/views/breadcrumbs/index.vue (hunk 2 above)
Comment (indirectlylit) [reply kolibri/core/assets/src/views/breadcrumbs/index.vue:49]: see note above re: `aria-hidden`

File: kolibri/core/assets/src/views/breadcrumbs/index.vue [hunk 5]
```diff
@@ -117,56 +137,53 @@
...
       },
 
-      resetCollapsedState() {
-        this.crumbs = this.crumbs.map(crumb => {
-          const updatedCrumb = crumb;
-          updatedCrumb.collapsed = false;
-          return updatedCrumb;
+      detachSensors() {
+        this.crumbs.forEach(crumb => {
+          crumb.sensor.detach();
```
Comment (indirectlylit) [inline kolibri/core/assets/src/views/breadcrumbs/index.vue]: this may not work as intended. I think you need to pass in a reference to the event you're detaching: https://github.com/marcj/css-element-queries/blob/master/src/ResizeSensor.js#L190

Looking at this more carefully, I think I'm doing it wrong in `responsive-element.js` by passing in both the element and the event.

File: kolibri/core/assets/src/views/breadcrumbs/index.vue [hunk 6]
```diff
@@ -117,56 +137,53 @@
             const updatedCrumb = crumb;
             updatedCrumb.ref = crumbRefs[index];
             updatedCrumb.sensor = new ResizeSensor(updatedCrumb.ref, () => {
-              this.throttleUpdateCrumbs();
+              this.updateCrumbs();
```
Comment (indirectlylit) [inline kolibri/core/assets/src/views/breadcrumbs/index.vue]: There might still be value in calling the throttled version. However, you could potentially lower the throttle interval to e.g. 100ms

File: kolibri/core/assets/src/views/breadcrumbs/index.vue [hunk 7]
```diff
@@ -69,23 +90,22 @@
...
-          crumbs.pop();
           // All, but the last, must have a valid router link
-          return crumbs.every(crumb => ValidateLinkObject(crumb.link));
+          return crumbItems.slice(0, -1).every(crumb => ValidateLinkObject(crumb.link));
         },
       },
     },
 
     data: () => ({
+      // will contain crumb items with their ref and collapsed state
```
Comment (indirectlylit) [inline kolibri/core/assets/src/views/breadcrumbs/index.vue]: it's an object with what attributes?

File: kolibri/core/assets/src/views/breadcrumbs/index.vue (hunk 3 above)
Comment (christianmemije) [reply kolibri/core/assets/src/views/breadcrumbs/index.vue]: If I use `v-show`, there's an issue that occurs if you open the drop down menu and the resize the screen, the drop down button is hidden but the drop down menu still is a bit visible.

File: kolibri/core/assets/src/views/breadcrumbs/index.vue [hunk 8]
```diff
@@ -117,56 +137,53 @@
...
         });
       },
 
       updateCrumbs() {
-        // reset collapsed values
-        this.resetCollapsedState();
+        const tempCrumbs = Array.from(this.crumbs);
+        let lastCrumbWidth = Math.ceil(tempCrumbs.pop().ref[0].getBoundingClientRect().width);
+        let remainingWidth = this.parentWidth - DROPDOWN_BTN_WIDTH - lastCrumbWidth;
+        const trackingIndex = this.crumbs.length - 2;
```
Comment (christianmemije) [inline kolibri/core/assets/src/views/breadcrumbs/index.vue]: Never decremented. Embarrassing.

**Comment by indirectlylit** [comment]
still gotta just make sure the events are detached: https://github.com/learningequality/kolibri/pull/1690/files#r123390293

otherwise looks good

**Comment by indirectlylit** [comment]
can simply do:

```javascript
updatedCrumb.sensor = new ResizeSensor(updatedCrumb.ref, this.throttledUpdateCrumbs);
```

and to detach

```javascript
crumb.sensor.detach(this.throttledUpdateCrumbs);
```

**Comment by rtibbles** [comment]
Will merge when tests pass.
